admin/class-phpredis-purger.php

```diff
@@ -39,11 +39,32 @@ public function __construct() {
 		try {
 
 			$this->redis_object = new Redis();
+			
+			$redis_acl = array();
+			
+			$username = $nginx_helper_admin->options['redis_username'];
+			$password = $nginx_helper_admin->options['redis_password'];
+			
+			if( $username && $password ) {
+				$redis_acl['auth'] = array( $username, $password );
+			}
+			
+			$hostname = empty( $nginx_helper_admin->options['redis_unix_socket'] ) ? $nginx_helper_admin->options['redis_hostname'] : $nginx_helper_admin->options['redis_unix_socket'];
+			$port     = empty( $nginx_helper_admin->options['redis_unix_socket'] ) ? $nginx_helper_admin->options['redis_port'] : 0;
+			
 			$this->redis_object->connect(
```

Comment:
On my local setup, there’s no Nginx server or Redis installed. When I switch branches from master to your branch, the code throws a fatal error. The plugin is trying to connect, but the connection is being refused. If the plugin is attempting to make a connection, there should also be a fatal error in the master branch.

Please take a look into this

Reply:
@SH4LIN I am unable to replicate the issue on my end. Can you please share the error here if possible?

Reply:
Sure.

Reply:
We are throwing a fatal error when the Redis connection is refused, which is understandable. However, this makes the entire admin side inaccessible. As a result, there is no way to update the connection configuration except by manually updating the value in the database or using WP-CLI.

One solution is to wrap the `$this->redis_object->connect` inside try catch and on the catch display WP notice

![image](https://github.com/user-attachments/assets/95b4d377-124d-472e-b634-49405f475bb3)

Reply:
@SH4LIN I found where the issue was and have fixed it. In predis it does not have checks for connection and instead immeditely executes the statement without checking if a connection is already established or not.

So in class-predis.php where we have addded selecting the databse feature, it was selecting the database without checking for connection establishment causing it to throw error.